File: src/main/java/at/yawk/fiction/impl/fanfiction/FfnSearchOrder.java
package at.yawk.fiction.impl.fanfiction;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum FfnSearchOrder {
    UPDATE_DATE("Update Date", 1),
    PUBLISH_DATE("Publish Date", 2),
    REVIEWS("Reviews", 3),
    FAVORITES("Favorites", 4),
    FOLLOWS("Follows", 5);

    private final String name;
    private final int id;
}

File: src/main/kotlin/at/yawk/fiction/impl/fanfiction/FfnSearchOrder.kt
package at.yawk.fiction.impl.fanfiction

enum class FfnSearchOrder(
        val displayName: String,
        val id: Int
) {
    UPDATE_DATE("Update Date", 1),
    PUBLISH_DATE("Publish Date", 2),
    REVIEWS("Reviews", 3),
    FAVORITES("Favorites", 4),
    FOLLOWS("Follows", 5),
}
